ts/RootContainer.tsx

```diff
@@ -34,13 +65,52 @@ const styles = StyleSheet.create({
  */
 class RootContainer extends React.Component<Props> {
   public componentDidMount() {
-    AppState.addEventListener("change", this.onApplicationActivityChange);
+    if (Platform.OS === "android") {
+      Linking.getInitialURL()
+        .then(this.navigate)
+        .catch(console.error);
+    } else {
+      Linking.addEventListener("url", this.handleOpenURL);
+    }
+
+    AppState.addEventListener("change", this.handleApplicationActivityChange);
   }
 
   public componentWillUnmount() {
-    AppState.removeEventListener("change", this.onApplicationActivityChange);
+    Linking.removeEventListener("url", this.handleOpenURL);
+    AppState.removeEventListener(
+      "change",
+      this.handleApplicationActivityChange
+    );
   }
 
+  private handleOpenURL = (event: { url: string }) => {
```

Comment:
use a consistent style for class methods, stick to `public componentWillUnmount() {}` style

Reply:
I used a fat arrow because this method has to be bound to `this`.